logstash-core/lib/logstash/api/lib/app/service.rb

```diff
@@ -40,4 +44,15 @@ def get(key)
     end
     LogStash::Json.dump(data)
   end
+
+  private
+
+  def has_counters?
```

Comment:
What I did in my test is to use the generator and custom output with an accessor for the number of events received.
So I wait until the output has received all the events generated by the generator.